server/balancer.go

```diff
@@ -771,3 +773,19 @@ func (h *balanceHotRegionScheduler) GetStatus() *StoreHotRegionInfos {
 		AsLeader: asLeader,
 	}
 }
+
+type cacheFilter struct {
+	cache *idCache
+}
```

Comment:
Not able to move this one due to idCache dependency problem. I'll deal with it in next PR.